### PR TITLE
feat: add vim resources

### PR DESCRIPTION
**Pull request by dagenius007**
## What does this PR do?
Add resource(s)

## For resources
Added vim book , [learn vim](https://www.youtube.com/playlist?list=PLm323Lc7iSW_wuxqmKx_xxNtJC_hJbQ7R)

### Why is this valuable (or not)?
adds resource to vim

### How do we know it's really free?
It is on youtube

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
Our other lists use the category name "IDE and Editors". For consistency, we should use that.

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -48,6 +48,7 @@
 * [HTML and CSS](#html-and-css)
     * [Bootstrap](#bootstrap)
 * [iOS](#ios)
+ [IDE and editors](#ide-and-editors)
```
Comment (eshellman) [inline courses/free-courses-en.md:51]:
```suggestion
* [IDE and editors](#ide-and-editors)
```

**Comment by dagenius007** [comment]
@eshellman this as been updated.

File: courses/free-courses-en.md (hunk 1 above)
Comment (eshellman) [reply courses/free-courses-en.md:51]: still need to fix

Comment (dagenius007) [reply courses/free-courses-en.md]: updated

**Pull request merged**